src/evaluate.py

```diff
@@ -0,0 +1,128 @@
+
+input_shape = (opt.channels, opt.img_height, opt.img_width)
+
+# Dimensionality (channel-wise) of image embedding
+shared_dim = opt.dim * 2 ** opt.n_downsample
+
+# Initialize generator and discriminator
+encoder = Encoder(dim=opt.dim, in_channels=opt.channels, n_downsample=opt.n_downsample)
+shared_G = ResidualBlock(features=shared_dim)
+G1 = Generator(dim=opt.dim, out_channels=opt.channels, n_upsample=opt.n_downsample, shared_block=shared_G)
+G2 = Generator(dim=opt.dim, out_channels=opt.channels, n_upsample=opt.n_downsample, shared_block=shared_G)
+
+if cuda:
+    encoder = encoder.cuda()
+    G1 = G1.cuda()
+    G2 = G2.cuda()
+
+assert os.path.exists("saved_models/%s/encoder_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained encoder exists
+assert os.path.exists("saved_models/%s/G1_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained G1 exists
+assert os.path.exists("saved_models/%s/G2_%02d.pth" % (opt.model_name, opt.epoch))  # check that trained G2 exists
+    
+# Load pretrained models
+encoder.load_state_dict(torch.load("saved_models/%s/encoder_%02d.pth" % (opt.model_name, opt.epoch)))
+G1.load_state_dict(torch.load("saved_models/%s/G1_%02d.pth" % (opt.model_name, opt.epoch)))
+G2.load_state_dict(torch.load("saved_models/%s/G2_%02d.pth" % (opt.model_name, opt.epoch)))
+
+# Set to eval mode 
+encoder.eval()
+G1.eval()
+G2.eval()
+
+Tensor = torch.cuda.FloatTensor if cuda else torch.Tensor
+dataloader = torch.utils.data.DataLoader(
+	DataProc(opt),
+	batch_size=opt.batch_size,
+	shuffle=True,
+	num_workers=opt.n_cpu,
+    pin_memory=True
+)
+
+feats = defaultdict(list)
+
+# ----------
+#  Testing
+# ----------
+
+progress = tqdm(enumerate(dataloader),desc='',total=len(dataloader))
+for i, batch in progress:
+
+    # Set model input
+    X1 = Variable(batch["A"].type(Tensor))
+    X2 = Variable(batch["B"].type(Tensor))
+
+    # -------------------------------
+    #  Infer with Encoder and Generators
+    # -------------------------------
+
+    # Get shared latent representation
+    mu1, Z1 = encoder(X1)
+    mu2, Z2 = encoder(X2)
+
+    # Reconstruct speech
+    recon_X1 = G1(Z1)
+    recon_X2 = G2(Z2)
+
+    # Translate speech
+    fake_X1 = G1(Z2)
+    fake_X2 = G2(Z1)
+
+    # Cycle translation
+    mu1_, Z1_ = encoder(fake_X1)
+    mu2_, Z2_ = encoder(fake_X2)
+    cycle_X1 = G1(Z2_)
+    cycle_X2 = G2(Z1_)
+        
+    # Plot batch every couple batch intervals
+    if i % opt.plot_interval == 0:
+        plot_batch_eval(opt.model_name, 'plot_A2B', i, X1, recon_X1, fake_X2)
+        plot_batch_eval(opt.model_name, 'plot_B2A', i, X2, recon_X2, fake_X1)
+        
+    # Append batch output to features dictionary
+    feats['A2B'].append([spect for spect in to_numpy(fake_X2)])
+    feats['B2A'].append([spect for spect in to_numpy(fake_X1)])
+        
+# Save converted output in pickle format
+pickle.dump(feats,open('out_eval/%s/out.pickle'%(opt.model_name),'wb'))
```

Comment:
probably you will need to mention the epoch number used to generate this output.

Reply:
Good idea